Drop the use of the deprecated setuptools Features

jamadden opened the pull request:
https://github.com/pypa/setuptools/issues/65

jamadden committed:
Drop the use of the deprecated setuptools Features

mgedmin commented on setup.py:
```diff
@@ -22,39 +22,37 @@
 }
 
+ext_modules = []
 if not is_pypy and not is_jython:
```
Maybe please add an `and not is_pure` check with the os.getenv('PURE_PYTHON') logic copied from zope.interface etc.? So people can ask to skip extension modules.

mgedmin replied:
You know what, you're right and I was wrong.  Build-time checking of PURE_PYTHON will poison pip's wheel cache and we should fix zope.interface not to do that.

jamadden replied:
I was slightly surprised to discover that you get different wheel file names depending on whether PURE_PYTHON is set (`zope.interface-4.7.2.dev0-py3-none-any.whl`) or not (`zope.interface-4.7.2.dev0-cp38-cp38-macosx_10_15_x86_64.whl`). I suspect it still messes with the cache though.

mgedmin commented on setup.py:
```diff
@@ -141,5 +133,11 @@ def _unavailable(self, e):
+    ext_modules=ext_modules,
+    extras_require={
+        'testing': [
+            'nose', 'coverage'
```
Do we actually allow `nose` as a test runner here?  AFAIU nose is deprecated upstream.

mgedmin commented on setup.py:
```diff
@@ -69,20 +67,14 @@ class optional_build_ext(build_ext):
-            # Python 2.5 and 3.x, which is needed in setup.py.
-            self._unavailable(sys.exc_info()[1])
+        except (CCompilerError, DistutilsExecError) as e:
+            self._unavailable(e)
```
:+1:

jamadden committed:
Update tests_require and 'test' extra.

Don't install nose, upstream no longer supports. We were actually testing with zope.testrunner on travis and in tox.ini so use that.

make appveyor do the same.

mgedmin commented on tox.ini:
```diff
@@ -5,7 +5,6 @@ envlist =
 [testenv]
 deps =
      .[test]
-     zope.testrunner
```
`extras = test` would be better than `deps = .[test]`

jamadden replied:
I knew there was another way to spell that but I always forget what it is.

mgedmin reviewed (approved):
LGTM!

jamadden committed:
Use extras=test in tox.ini. [skip ci]

Per https://github.com/zopefoundation/zope.i18nmessageid/pull/19#discussion_r368547455 from @mgedmin
